arrows/vxl/convert_image.cxx

```diff
@@ -334,9 +334,16 @@ ::scale_and_convert( vil_image_view< ipix_t > const& input )
     percentile_scale_image(
       input, output, percentile_norm, 1.0 - percentile_norm, 1e8 );
   }
-  else if( scale_factor == 0.0 )
+  else if( scale_factor == 0.0 || scale_factor == 1.0 )
   {
-    vil_convert_cast( input, output );
+    if( std::is_same< opix_t, ipix_t >::value )
```

Comment:
no harm in this, but `vil_convert_cast` already does a similar optimization.  Just FYI.

Reply:
Ah... my IDE unhelpfully steered me toward [L728](https://github.com/vxl/vxl/blob/7edc7cd869ac08a2d236e55ec173ae3c09096fa5/core/vil/vil_convert.h#L728) rather than the correct [L245](https://github.com/vxl/vxl/blob/7edc7cd869ac08a2d236e55ec173ae3c09096fa5/core/vil/vil_convert.h#L245)...